client/src/components/ProductCard/ProductCard.js

```diff
@@ -14,11 +25,36 @@ function ProductCard({ params }) {
         </Card.Body>
       </Link>
       <Card.Footer>
-        <small className="text-muted">
-          <Moment format="d MMM YYYY (dddd) HH:mm">{params.addedAt}</Moment>-{" "}
-          <strong>{params.city}</strong>
-          {/* <Link to="" id="heartIcon"><BsHeart /></Link> */}
-        </small>
+        {params?.city && (
+          <OverlayTrigger
+            show={!locationIsOverflowing ? false : undefined}
+            placement="top"
+            overlay={
+              <Tooltip id={`tooltip-${params.addedAt}`}>{params.city}</Tooltip>
+            }
+          >
+            <small className="text-muted text-ellipsis" ref={locationRef}>
+              <strong>Location: </strong>
+              {params.city}
+            </small>
+          </OverlayTrigger>
+        )}
+        {params?.addedAt && (
+          <OverlayTrigger
+            show={!dateIsOverflowing ? false : undefined}
+            placement="top"
+            overlay={
+              <Tooltip id={`tooltip-${params.addedAt}`}>
+                <Date>{params.addedAt}</Date>
+              </Tooltip>
+            }
+          >
+            <small className="text-muted text-ellipsis" ref={dateRef}>
```

Comment:
To resolve issue 1, I refactored the footer lines into the existing small element, although I wouldn't recommend using small because there are 2 items handling styles, the html and the added css.

Recommend consolidating to CSS styles.